core/src/conn.rs

```diff
@@ -17,19 +17,136 @@ pub struct MongoConnection {
 }
 
 impl MongoConnection {
-    // Creates a new MongoConnection with the given settings and execute a dummy aggregation to
-    // validate it works.
-    // The operation will timeout if it takes more than loginTimeout seconds.
+    // Creates a new MongoConnection with the given settings and lists databases to make sure the
+    // connection is legit.
+    //
+    // The operation will timeout if it takes more than loginTimeout seconds. This timeout is
+    // delegated to the mongo rust driver.
+    //
     // The initial current database if provided should come from SQL_ATTR_CURRENT_CATALOG
     // and will take precedence over the database setting specified in the uri if any.
-    // The initial operation time if provided should come from  and will take precedence over the
+    // The initial operation time if provided should come from and will take precedence over the
     // setting specified in the uri if any.
     pub fn connect(
```

Comment:
I am not sure about the separation between the odbc and core layers here. I thought that we wanted the core package to be a thin wrapper around the Rust driver and not hold much ODBC specific knowledge.
Because of that I was expecting that `connect` here could be used by all the SQLxxConnect methods when/if we add more.
Also, I think that dealing with loading DSN info from registry or INI files is definitely specific to ODBC, and I would put it in the odbc package.
On way to do this would be:
- In the odbc package, do the gymnastic you have here to transform ODBC connection setting info (either DSN info or direct connection strings plus any SQLxxConnect arguments like UID/PWD for example) to a MongoDB uri. The code can/should be in helpers so that all SQLxxConnect functions can use it.
- Keep the code you have from line 58 onward, and add the update of the value for database, connectTimeoutMS using current_db and login_timeout.

This way the code would be ready for us if/when we had more SQLxxConnect methods.
What do you think?

Reply:
Ok, this is interesting to me, because my idea for the core api was too be an ODBC driver except for the parts that are specific to C (i.e., do as much as possible in the core api). I'm not sure on the best way to go.

Reply:
We discussed, the parsing for uris will be added to odbc crate and the name of `uri` in `connect` (in core) will be changed to `mongo_uri` to make things more clear